src/when.js

```diff
@@ -0,0 +1,33 @@
+var _curry2 = require('./internal/_curry2');
+var identity = require('./identity');
+var ifElse = require('./ifElse');
+
+
+/**
+ * Creates a function that will first pass its argument to the given predicate
+ * function. If the predicate returns true, the function will return the result
+ * of calling the `whenTrue` function with the same argument. If the predicate
+ * returns false, the argument is returned as is.
+ *
+ * @func
+ * @memberOf R
+ * @category Logic
+ * @see R.ifElse, R.unless
+ * @sig (a -> Boolean) -> (a -> a) -> (a -> a)
+ * @param {Function} condition A predicate function
+ * @param {Function} whenTrue  A function to invoke when the `condition`
+ *                             evaluates to a truthy value.
+ * @return {Function} A new function that will process the `whenTrue` function
+ *                    depending upon the result of the `condition` predicate.
+ * @example
+ *
+ *      var truncate = R.when(
```

Comment:
Although we don't usually put them in examples, I think here it might be worth including the signature of `truncate`.